plugins/module_utils/vmware.py

```diff
@@ -218,3 +224,141 @@ def is_vcenter(self):
             return True
         elif api_type == 'HostAgent':
             return False
+
+    def __get_objs_by_name(self, vimtype, name, first_only=False):
+        """
+        Get any vsphere object associated with a given text name and vim type.
+        Args:
+            vimtype: The type of object to search for
+            name: The name of the object to search for
+            first_only: If true, return once the first object is found.
+                        Useful when names must be unique
+        Returns:
+            list(object) or list() if no matches are found
+        """
+        objs = []
+        container = self.content.viewManager.CreateContainerView(self.content.rootFolder, vimtype, True)
+        for c in container.view:
+            if c.name == name:
+                if first_only:
+                    return c
+                objs += [c]
+
+        return objs if objs else None
+
+    def get_vm_by_name(self, vm_name, fail_on_missing=False, name_match=None):
+        """
+            Get the vms matching the given name. VM names may not be unique
+            in a given cluster.
+            Args:
+                vm_name: Name of the VM to search for
+                fail_on_missing: If true, an error will be thrown if no VMs are found
+                name_match: If provided, return one VM from the list of VMs. Either 'first' or 'last'
+            Returns:
+                list(vm), vm if name_match is provided, or None if no matches were found
```

Comment:
It may also return just string in case `name_match` is provided. Might be bit confusing. I think better to return always a list or a string.

Reply:
i think it should be a list of VM objects instead of just a string since the object is generally more useful. But always returning a list does sound less confusing, ill implement that!